src/library_webgl2.js

```diff
@@ -695,24 +697,20 @@ var LibraryWebGL2 = {
   },
 
   glClientWaitSync__sig: 'iiiii',
+  glClientWaitSync__deps: ['$convertI32PairToI53'],
   glClientWaitSync: function(sync, flags, timeoutLo, timeoutHi) {
     // WebGL2 vs GLES3 differences: in GLES3, the timeout parameter is a uint64, where 0xFFFFFFFFFFFFFFFFULL means GL_TIMEOUT_IGNORED.
     // In JS, there's no 64-bit value types, so instead timeout is taken to be signed, and GL_TIMEOUT_IGNORED is given value -1.
     // Inherently the value accepted in the timeout is lossy, and can't take in arbitrary u64 bit pattern (but most likely doesn't matter)
     // See https://www.khronos.org/registry/webgl/specs/latest/2.0/#5.15
-    timeoutLo = timeoutLo >>> 0;
-    timeoutHi = timeoutHi >>> 0;
-    var timeout = (timeoutLo == 0xFFFFFFFF && timeoutHi == 0xFFFFFFFF) ? -1 : makeBigInt(timeoutLo, timeoutHi, true);
-    return GLctx.clientWaitSync(GL.syncs[sync], flags, timeout);
+    return GLctx.clientWaitSync(GL.syncs[sync], flags, convertI32PairToI53(timeoutLo, timeoutHi));
```

Comment:
this used to be unsigned, does it not need to still be? (likewise below)

Reply:
Yeah, I think not - the old code was "treat unsigned, but -1 is signed". I don't think that was the spirit of the code, but it was just an uninformed way to get the -1 handling proper. To me it seems more appropriate to always treat this signed.